aws_signing_helper/pkcs11_signer.go

```diff
@@ -110,64 +270,157 @@ func getCerts(module *pkcs11.Ctx, session pkcs11.SessionHandle) (certs []*x509.C
 		crtAttributes := []*pkcs11.Attribute{
 			pkcs11.NewAttribute(pkcs11.CKA_VALUE, 0),
 		}
-
 		if crtAttributes, err = module.GetAttributeValue(session, certObject, crtAttributes); err != nil {
 			return nil, err
 		}
 
 		rawCert := crtAttributes[0].Value
-		curCert, err := x509.ParseCertificate(rawCert)
+
+		var cert_obj CertObjInfo
+
+		cert_obj.x509, err = x509.ParseCertificate(rawCert)
 		if err != nil {
 			return nil, errors.New("error parsing certificate")
 		}
 
-		certs = append(certs, curCert)
+		// If we have any *additional* criteria in 'certIdentifier', drop
+		// any certificates which don't match. In the common case there
+		// not additional criteria and certMatches() matches everything.
+		// (Most implementations won't want that additional filter; in
```

Comment:
TBH, I don't think we necessarily need to allow for the `--cert-selector` flag to work when a user leverages the PKCS#11 integration since, as you said, the URI should be sufficient. We use `--cert-selector` for identifying certificates and their associated keys within OS certificate stores though.